### PR TITLE
Fix getter methods for Java model with additionalProperties

### PR DESCRIPTION
**Pull request by stianlik**
See [merged PR for swagger-codegen](https://github.com/swagger-api/swagger-codegen/pull/8245)

### CODE REVIEW

**Comment by wing328** [comment]
cc @bbdouglas (2017/07) @JFCote (2017/08) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) for review

File: samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/JSON.java [hunk 1]
```diff
@@ -64,7 +64,8 @@ public Class getClassForElement(JsonElement readElement) {
             }
           })
         ;
-        return fireBuilder.createGsonBuilder();
+        GsonBuilder builder = fireBuilder.createGsonBuilder();
```
Comment (JFCote) [inline samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/JSON.java]: why this change?

File: samples/client/petstore/java/rest-assured/src/main/java/org/openapitools/client/JSON.java [hunk 2]
```diff
@@ -64,7 +64,8 @@ public Class getClassForElement(JsonElement readElement) {
             }
           })
         ;
-        return fireBuilder.createGsonBuilder();
+        GsonBuilder builder = fireBuilder.createGsonBuilder();
```
Comment (JFCote) [inline samples/client/petstore/java/rest-assured/src/main/java/org/openapitools/client/JSON.java]: why?

**Comment by jmini** [comment]
> Shippable — Run 770 status is FAILED. 

It seems that other java clients under `samples/` are affected by this change.

* run `mvn clean verify` locally
* run `bin/java-petstore-all.sh`
* commit changes to this PR.

(instead of `bin/java-petstore-all.sh` you can also run `bin/ensure-up-to-date` -- the same script than the one executed by Shippable CI. But this will take longer to run).

File: samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/JSON.java (hunk 1 above)
Comment (stianlik) [reply samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/JSON.java]: This is a change related to one of my other PRs, not sure why it was included here, removed after regenerating samples. Sorry about that.

**Comment by stianlik** [comment]
It looks like this issue has been resolved differently in openapi-generator. When `additionalProperties` is defined, the java client does not implement a domain class, but returns a generic `Map` instead. If that is the case, this PR is useless.

@jmini @JFCote Can you confirm that the expected behavior is as described above?

**Comment by JFCote** [comment]
I have create a bug for this here:
https://github.com/OpenAPITools/openapi-generator/issues/191

**Comment by stianlik** [comment]
I'm closing this PR as it's a bugfix for an obsolete solution to #191. Assuming #191 is solved properly, it is no need for an additional bugfix.
